### PR TITLE
Documentation updates

### LINKED ISSUE #97: Missing exposed port
No port is exposed by default, making the use of docker-compose-letsencrypt-nginx-proxy-companion a necessity and not optional.

Either port 80 should be exposed or usage of docker-compose-letsencrypt-nginx-proxy-companion integrated within current docker-compose files.

I lost about 4 hours figuring this basic thing out, the documentation is not clear.

### PR DESCRIPTION
**Pull request by ddavidebor**
I've rewritten and updated the documentation

Fixes #97 
Depends on #100

### CODE REVIEW

**Comment by revant** [comment]
I'll suggest we move docs to gh-pages branch.

It triggers a build on change of docs

**Comment by ddavidebor** [comment]
gh-pages is used usually for web pages rendered from the docs, not for the docs themselves

**Comment by Thunderbottom** [comment]
David, I think there's a lot of confusion right now. This docker that we have right now clearly isn't meant for development purpose. We can have one later, but at the moment we'd like to ease deployments by pushing people to setup their instances using docker for production use.

There are also some issues with the documentation that you've changed. 

We also don't want to just merge documentation right away since it'll trigger another build on travis causing a new build to be pushed to dockerhub. I think we'll just move the entire documentation off to some other page, maybe gh-pages sounds fine for that too. So as long as a new build isn't triggered by such a trivial change, we'll be fine.

I am thinking of having the master branch frozen for pull requests and instead have another staging branch which accepts pull requests and doesn't run travis builds, while only master branch would push images to dockerhub. I'll write documentation on it in a while.

**Comment by revant** [comment]
can you close this and send it to develop branch?

I just made https://github.com/frappe/frappe_docker/wiki/Contribution-Guidelines

Merge

**Comment by ddavidebor** [comment]
@Thunderbottom no worries it's clear now. I've changed the base to the develop branch, will change to whatever you prefer accordingly.
Please comment on the documentation if you'd like something amended.

I've been waiting for frappe to be production-ready on docker for a long time, so I'm more than happy to help.
I understand this repository is for production, but there was a repository for development here last time I checked, this was changed into a production repository and the previous one is not clearly accessible anymore.

We should either add the develop features to this one or branch out and document the previous one so that it's still accessible.

You should definitely freeze the master banch as soon as the project gets used in production.

File: README.md [hunk 1]
```diff
@@ -68,13 +48,70 @@ To get started, copy the existing `env-example` file to `.env` inside the `insta
...
+    -f installation/docker-compose-common.yml \
+    -f installation/docker-compose-frappe.yml \
+    -f installation/docker-compose-networks.yml \
+    --project-directory installation run --publish 80:80 frappe-nginx
+```
+
+Make sure to replace `<project-name>` with any desired name you wish to set for the project.
+
+Note: this command does not run docker-compose in daemon mode. You will need to keep the terminal window open.
+Note: the local deployment is for testing only. The site names are limited to patterns matching *.localhost.
```
Comment (Thunderbottom) [inline README.md]: needs escaping I think
```suggestion
Note: the local deployment is for testing only. The site names are limited to patterns matching \*.localhost.
```

File: README.md [hunk 2]
```diff
@@ -68,13 +48,70 @@ To get started, copy the existing `env-example` file to `.env` inside the `insta
...
 - `SITES=site1.domain.com,site2.domain.com`
     - List of sites that are part of the deployment "bench". Each site is separated by a comma(,).
-    - If LetsEncrypt is being setup, make sure that the DNS for all the site domains are pointing to the current instance.
+    - If LetsEncrypt is being setup, make sure that the DNS for all the site's domains are pointing to the current instance.
 - `LETSENCRYPT_EMAIL=your.email@your.domain.com`
-    - Email for LetsEncrypt expiry notification. This is only required if you are setting up the nginx proxy companion.
+    - Email for LetsEncrypt expiry notification. This is only required if you are setting up LetsEncrypt.
+
+
+### Deployment for local development
```
Comment (Thunderbottom) [inline README.md]: I think the current readme is pretty good else for the grammatical and the spacing changes that you have made.

Comment (ddavidebor) [reply README.md]: Most of the changes are reorganizing it so it's clearer and adding new sections, grammar and spacing is just a byproduct.

File: README.md (hunk 1 above)
Comment (ddavidebor) [reply README.md]: Thank you

File: README.md (hunk 2 above)
Comment (ddavidebor) [reply README.md]: I had troubles understanding the original readme, wasted many hours of work, all the details that were unclear to me (including those for which I opened issues) were clarified.

Comment (Thunderbottom) [reply README.md]: But this section shouldn't exist. There's no "Deployment for local development" that is being provided through this docker. It's better to not misguide people on that. I hope it is understandable.

**Comment by revant** [comment]
I've added back development container #113 

Please pull latest so that only relevant files are in diff.

**Comment by ddavidebor** [comment]
@revant diff cleaned up

File: README.md (hunk 2 above)
Comment (revant) [reply README.md]: These set of production images can be setup locally to make api calls from local non frappe app during development.

That needs port 80 to be exposed and makes the documentation relevant

To clear confusion may be mention somewhere about development/README.md

**Pull request merged**